File: JavaPractice/src/Solution9.java
public class Solution9 {
    public boolean isPalindrome(int x) {
        if (x < 0) return false;

        long working_value = x;
        long reversed = 0;

        while (working_value != 0) {
            reversed *= 10;
            reversed += working_value % 10;
            working_value /= 10;
        }

        return (x == reversed);
    }
}

File: CppPractice/Solution9.cpp
//
// Created by Germain Jones on 30/07/2025.
//

#include "Solution9.h"

#include <iostream>

bool Solution9::isPalindrome(int x) {
    if (x < 0) return false;

    int working_value = x;
    int reverse_value = 0;

    while (working_value != 0) {
        reverse_value *= 10;
        reverse_value += working_value % 10;
        working_value /= 10;
    }

    return x == reverse_value;
}
